### PR TITLE
Attempt to fix builds on Travis

### PR DESCRIPTION
**Pull request by nicoddemus**
* Drop Python 3.5: conda forge no longer supports it, so packages in there are getting old
* Get most dependencies from conda-forge
* Fix linting max-line length to more reasonable lengths we use internally

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,7 +7,6 @@ env:
     - SCRIPT=lint
     - SCRIPT=tests
 python:
-  - "3.5"
```
Comment (prusse-martin) [inline .travis.yml]: Why did you remove 3.5?

Comment (nicoddemus) [reply .travis.yml]: * We no longer have up to date packages for 3.5 on conda-forge
* We don't use Python 3.5 anymore, so no reason to keep supporting it on our side

Our idea was to "keep supporting until it doesn't cost us anything to do so", which is no longer the case really. 🤷‍♂
